### ISSUE TITLE
Allow display in ViewStylePropTypes?

### ISSUE THREAD

**Issue by tuckerconnelly** [necolas/react-native-web#296]
Sometimes I need to hack the style.display of a View.

Here's an example where I set display to `table-cell`:

https://github.com/tuckerconnelly/carbon-ui/blob/master/src/components/DataTable/Cell.js

**Comment by necolas** [comment]
Why do you need to set `table-cell` there?

FYI, there is `Platform.select` in React Native. And rather than `[styles.base].concat(style)` you can simply have `[styles.base, style]`.

**Comment by tuckerconnelly** [comment]
Thanks :D

Maybe this is a super edge case, but flexbox in react-native behaves more like `display: table`, but behaves like normal flexbox on web.

I was thinking I'd hack something like:

```
import ViewStylePropTypes from 'rnw/wherever'

ViewStylePropTypes.display = PropTypes.oneOf(...)
``

if you wanted to keep them pure (understandable).

**Comment by dorthwein** [comment]
Side note - I ran into an issue with a react-native component that used View.propTypes.style and resulted in breaking app.  I submitted a pull request to that other project but as a side - allowing access to this would help cross platform package compatibility.

**Comment by necolas** [comment]
I don't understand that last comment

**Comment by dorthwein** [comment]
If you look at react-native-flip-card they make a call to View.propTypes.style but it returns undefined if your using redact-native-web thus breaking. Maybe I'm miss understanding what is being proposed but If you add access to ReactNative.View.propTypes - it would resolve that breakin their package ;)

**Comment by necolas** [comment]
View.propTypes.style is already included as you can see in the source code. It will be undefined in production builds but that shouldn't be a problem in propTypes

**Comment by blairio** [comment]
Another reason to allow `display` in `ViewStylePropTypes` is to work around this issue in Safari https://bugs.webkit.org/show_bug.cgi?id=150445 which causes huge performance issues with nested flex boxes. I know it's not great, but given React Native Web's prolific use of flexbox it is a necessary evil and the errors that gets thrown in developement mode are very annoying.

Thanks!

**Comment by necolas** [comment]
I wouldn't be surprised if RN added support for `display` in the future. I'll remove the warnings for `display`. I'm hoping there is a way to auto-fix this within the framework (see #42) while webkit has this problem (not fixed for over 2 years :/). We have this same problem on mobile.twitter.com and insert a `div` into a tree where it doesn't impact child layout - in RNW we might be able to infer that from context and/or child styles (may slow things down though).
